server/lib/spam.ts

```diff
@@ -0,0 +1,70 @@
+import slackLib from '../lib/slack';
+import activities from '../constants/activities';
+
+interface Collective {
+  id: string;
+}
+
+/**
+ * SpamControl: Singleton Design Pattern
+ */
+export class SpamControl {
+  private static instance: SpamControl;
+  private collectiveCheckList: string[] = ['name', 'website', 'description', 'longDescription'];
+  private blackList: string[] = ['keto', 'porn', 'pills'];
+
+  public static getInstance(): SpamControl {
+    if (!SpamControl.instance) {
+      SpamControl.instance = new SpamControl();
+    }
+
+    return SpamControl.instance;
+  }
+
+  /**
+   * Returns SuspiciousKeywords
+   * @param {String} content
+   */
+  private getSuspiciousKeywords(content: string): string[] {
+    const suspiciousWords = [];
+    this.blackList.forEach(keyword => {
+      if (content.includes(keyword)) {
+        suspiciousWords.push(keyword);
+      }
+    });
+    return suspiciousWords;
+  }
+
+  /**
+   *
+   * @param collective {Collective}
+   */
+  private collectiveCheck(collective: Collective): { collective: Collective; warnings: object } {
```

Comment:
Good idea to return a map for `warnings`, it's a good way to bundle all of them.

Why returning the collective? You pass it as a parameter so you already have it when you call this function.

Reply:
it's my bad, the structure is a bit different than this. Then, I moved the whole logic inside spam.js and call slack function inside. Now don't need to return anything. Resolved.